control/config.py

```diff
@@ -7,40 +7,143 @@
 
 
-class GatewayConfig:
-    """Loads and returns config file settings.
+class GatewaySubConfig(BaseModel):
+    """Validates config file settings and types for gateway.
 
-    Instance attributes:
-        config: Config parser object
+    Required fields report error on missing keys and on invalid values for 
+    present keys.
+
+    Required Fields:
+        name: Gateway identifier (default: local hostname)
+        group: Gateway group identifier
+        addr: Gateway server listener IP address
+        port: Gateway server listener port (1-65535)
+        enable_auth: Use mTLS for client-server authentication
+        state_update_notify: Enable watch/notify for updates
+        state_update_interval_sec: Polling interval for updates (>0s)
     """
-    def __init__(self, conffile):
-        with open(conffile) as f:
-            self.config = configparser.ConfigParser()
-            self.config.read_file(f)
+    name: str
+    group: str
+    addr: IPvAnyAddress
+    port: conint(ge=1, le=65535)
+    enable_auth: bool
+    state_update_notify: bool
+    state_update_interval_sec: PositiveInt
+
+    @validator('name')
+    def set_default_name(cls, name):
+        """Sets gateway name to the local hostname if unspecified in config."""
+        if not name:
+            return socket.gethostname()
+        return name
+
 
-    def get(self, section, param):
-        return self.config.get(section, param)
+class CephSubConfig(BaseModel):
+    """Validates config file settings and types for ceph.
 
-    def getboolean(self, section, param):
-        return self.config.getboolean(section, param)
+    Required fields report error on missing keys and on invalid values for 
+    present keys.
+
+    Required Fields:
+        pool: Ceph pool to store gateway state object
+        config_file: Ceph config file
+    """
+    pool: constr(min_length=1)
+    config_file: str
 
-    def getint(self, section, param):
-        return self.config.getint(section, param)
+    @validator('config_file')
+    def check_log_level(cls, config_file):
+        """Confirms the config file exists at specified path."""
+        if not os.path.isfile(Path(config_file)):
+            raise ValueError(f'Invalid Ceph config file path: {config_file}')
+        return config_file
 
-    def getfloat(self, section, param):
-        return self.config.getfloat(section, param)
 
-    def get_with_default(self, section, param, value):
-        return self.config.get(section, param, fallback=value)
+class MtlsSubConfig(BaseModel):
+    """Validates config file settings and types for MTLS.
+
+    Required fields report error on missing keys and on invalid values for 
+    present keys.
+
+    Required Fields:
+        server_key: Server key
+        client_key: Client key
+        server_cert: Server certificate
+        client_cert: Client certificate
+    """
+    server_key: str
+    client_key: str
+    server_cert: str
+    client_cert: str
```

Comment:
`argparse` already provides a [`FileType`](https://docs.python.org/3/library/argparse.html#filetype-objects)` that not only validates whether the file exists or not, or can be accessed, etc, but it automatically provides its contents.